### PR TITLE
deprecate mean_file option in Caffe

### PR DESCRIPTION
**Pull request by sadolini**
Root cause analysis: Deprecate option mean_file for Caffe, mean_values option can be used instead

Ticket: 67411

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR **NA**
* [x]  Transformation preserves original framework node names **NA**

Validation:
* [x]  Unit tests **NA**
* [x]  Framework operation tests **NA**
* [x]  Transformation tests **NA**
* [x]  Model Optimizer IR Reader check **NA**

Documentation:
* [x]  Supported frameworks operations list **NA**
* [x]  Guide on how to convert the **public** model **NA**
* [x]  User guide update **NA**

### CODE REVIEW

**Comment by achetver** [comment]
Looks good. Do we have any tests for this command line parameter which need to be updated?

**Review by vgavrilo — APPROVED** [review]

**Comment by sadolini** [comment]
> @sadolini Looks good. Do we have any tests for this command line parameter which need to be updated?

I have not found such. Anyway we just deprecate, don't remove functionality

**Review by achetver — APPROVED** [review]

**Review by yekruglov — APPROVED** [review]

File: docs/MO_DG/prepare_model/convert_model/Convert_Model_From_Caffe.md [hunk 1]
```diff
@@ -63,10 +63,12 @@ Caffe*-specific parameters:
   -k K                  Path to CustomLayersMapping.xml to register custom
                         layers
   --mean_file MEAN_FILE, -mf MEAN_FILE
-                        Mean image to be used for the input. Should be a
+                        [Deprecated option, please use mean_values option instead]
+						Mean image to be used for the input. Should be a
```
Comment (rkazants) [inline docs/MO_DG/prepare_model/convert_model/Convert_Model_From_Caffe.md:67]: identation is wrong. Please fix it. Also, I propose to write as follows:
```
[DEPRECATED] Mean image to be used for the input.
Should be a binary proto file. Please use --mean_values option instead.
please use mean_values option instead
```

File: docs/MO_DG/prepare_model/convert_model/Convert_Model_From_Caffe.md [hunk 2]
```diff
@@ -63,10 +63,12 @@ Caffe*-specific parameters:
...
                         binaryproto file
   --mean_file_offsets MEAN_FILE_OFFSETS, -mo MEAN_FILE_OFFSETS
-                        Mean image offsets to be used for the input
+                        [Deprecated option, please use mean_values option instead]
+						Mean image offsets to be used for the input
```
Comment (rkazants) [inline docs/MO_DG/prepare_model/convert_model/Convert_Model_From_Caffe.md:71]: The same comment. Please correct as mentioned above

File: model-optimizer/mo/utils/cli_parser.py [hunk 3]
```diff
@@ -89,12 +89,12 @@ def __call__(self, parser, namespace, values, option_string=None):
 
 class DeprecatedCanonicalizePathCheckExistenceAction(CanonicalizePathCheckExistenceAction):
     def __call__(self, parser, namespace, values, option_string=None):
-        super().__call__(parser, namespace, values, option_string)
         dep_msg = "Use of deprecated cli option {} detected. Option use in the following releases will be fatal. ".format(
             option_string)
         if 'tensorflow_use_custom_operations_config' in option_string:
             dep_msg += 'Please use --transformations_config cli option instead'
```
Comment (rkazants) [inline model-optimizer/mo/utils/cli_parser.py:95]: I think you need to add `--mean_file` and `--mean_offsets` and recommend alternatives

**Review by rkazants — COMMENTED** [review]
Also, please update other documentation where these options are mentioned. We need to remove them except FAQ while it is alive. I found Converting_model.md, Converting_General.md, and paddle

**Comment by sadolini** [comment]
> Also, please update other documentation where these options are mentioned. We need to remove them except FAQ while it is alive. I found Converting_model.md, Converting_General.md, and paddle

Fixed

File: model-optimizer/mo/utils/cli_parser.py [hunk 4]
```diff
@@ -89,12 +89,14 @@ def __call__(self, parser, namespace, values, option_string=None):
...
+        if 'mean_file' in option_string or 'mean_offset' in option_string:
+            dep_msg += 'Please use --mean_values cli option instead.'
         log.error(dep_msg, extra={'is_warning': True})
+        super().__call__(parser, namespace, values, option_string)
```
Comment (rkazants) [inline model-optimizer/mo/utils/cli_parser.py:99]: why is this changed? Please move to the preliminary code line

**Review by rkazants — COMMENTED** [review]
Looks good to me. Left one comment

File: model-optimizer/mo/utils/cli_parser.py (hunk 4 above)
Comment (sadolini) [reply model-optimizer/mo/utils/cli_parser.py:99]: Because if we have error inside option processing, warning about deprecation will be missed, I think it is wrong, user should see deprecation warning in any way.
